app/src/main/java/org/fossasia/phimpme/accounts/AccountActivity.java

```diff
@@ -326,6 +328,22 @@ public void onClick(DialogInterface dialog, int which) {
         }
     }
 
+    private void checkInstagram() {
+            PackageManager packageManager = (ActivitySwitchHelper.context).getPackageManager();
+            if(Utils.isAppInstalled(PACKAGE_INSTAGRAM,packageManager)){
+                realm.beginTransaction();
+                account = realm.createObject(AccountDatabase.class, INSTAGRAM.toString());
+                account.setAccountname(INSTAGRAM);
+                account.setUsername(INSTAGRAM.toString());
+                realm.commitTransaction();
```

Comment:
What does it mean to store instagram as a username ? Instagram does not provide any api then we don't in accounts and just add in accounts name and increase the below variable hiddenInAccounts. 
Remove check instagram function completely. Just add shareToInstagram() and put the share code in it. No need to check again is installed because instagram option onky visible if user is installed.